src/main/java/com/google/sps/model/AuthenticationVerifierImpl.java

```diff
@@ -23,11 +23,13 @@
 import java.io.IOException;
 import java.security.GeneralSecurityException;
 import java.util.Collections;
+import java.util.Optional;
 
 /** Verifies authentication information with Google */
 public class AuthenticationVerifierImpl implements AuthenticationVerifier {
   @Override
-  public boolean verifyUserToken(String idToken) throws GeneralSecurityException, IOException {
+  public Optional<String> getUserEmail(String idToken)
```

Comment:
Interesting data structure - I did not know this existed

Reply:
Yeah! Anton told me about it and I instantly loved it - I didn't know these existed outside of Swift!

Reply:
Nice use of Optional - I like how it help make the tests more readable too.